docs/content/topics/operators/gateways.md

```diff
@@ -11,340 +11,518 @@ aliases:
+default root user enabled. If you'd like to follow along and haven't yet
+deployed Brigade, check out the [QuickStart].
 
-First we can begin with a simplified version of the script above. This one does not
-run in a loop. It just runs once to completion.
+[QuickStart]: /intro/quickstart
 
-Here is `cron-event.sh`:
+### Authentication
 
-```bash
-#!/usr/bin/env bash
-set -euo pipefail
+As a first course of action for our example gateway written in Go, we'll need
+to authenticate with Brigade. Here is what the Go code looks like, with in-line
+comments describing each section:
 
-# The Kubernetes namespace in which Brigade is running.
-namespace=${NAMESPACE:-default}
+```go
+package main
 
-event_provider="simple-event"
-event_type="my_event"
-project_id="brigade-830c16d4aaf6f5490937ad719afd8490a5bcbef064d397411043ac"
-commit_ref="master"
-commit_id="589e15029e1e44dee48de4800daf1f78e64287c0"
-uuid="$(uuidgen | tr '[:upper:]' '[:lower:]')"
-name="simple-event-$uuid"
+import (
+  "context"
+  "log"
 
-payload=$(date)
-script=$(cat <<EOF
-const { events } = require("brigadier");
-events.on("my_event", (e) => {
-  console.log("The system time is " + e.payload);
-});
-EOF
+  brigadeOS "github.com/brigadecore/brigade-foundations/os"
+  "github.com/brigadecore/brigade/sdk/v2/authn"
+  "github.com/brigadecore/brigade/sdk/v2/restmachinery"
 )
 
-cat <<EOF | kubectl --namespace ${namespace} create -f -
-apiVersion: v1
-kind: Secret
-metadata:
-  name: ${name}
-  labels:
-    heritage: brigade
-    project: ${project_id}
-    build: ${uuid}
-    component: build
-type: "brigade.sh/build"
-data:
-  revision:
-    commit: $(base64 -w 0 <<<"${commit_id}")
-    ref: $(base64 -w 0 <<<"${commit_ref}")
-  event_provider: $(base64 -w 0 <<<"${event_provider}")
-  event_type: $(base64 -w 0 <<<"${event_type}")
-  project_id: $(base64 -w 0 <<<"${project_id}")
-  build_id: $(base64 -w 0 <<<"${uuid}")
-  payload: $(base64 -w 0 <<<"${payload}")
-  script: $(base64 -w 0 <<<"${script}")
-EOF
+func main() {
+  ctx := context.Background()
+
+  // Get the Brigade API server address and root user password from env
+  apiServerAddress, err := brigadeOS.GetRequiredEnvVar("APISERVER_ADDRESS")
+  if err != nil {
+    log.Fatal(err)
+  }
+  rootPassword, err := brigadeOS.GetRequiredEnvVar("APISERVER_ROOT_PASSWORD")
+  if err != nil {
+    log.Fatal(err)
+  }
+
+  // Create a root session to acquire an auth token
+
+  // The default Brigade deployment mode uses self-signed certs
+  // Hence, we allow insecure connections in our APIClientOptions
+  apiClientOpts := &restmachinery.APIClientOptions{
+    AllowInsecureConnections: true,
+  }
+
+  // Create a new auth client
+  authClient := authn.NewSessionsClient(
+    apiServerAddress,
+    "",
+    apiClientOpts,
+  )
+
+  // Create a root session, which returns an auth token
+  token, err := authClient.CreateRootSession(ctx, rootPassword)
+  if err != nil {
+    log.Fatal(err)
+  }
+  tokenStr := token.Value
+}
```

Comment:
I don't think this is a pattern I'd want people to copy (and people will). It might be better to show commands to generate a service account and source the service account's token from an env var.

Reply:
Good call.  Updated!